tests/e2e/svmotion_volumes.go

```diff
@@ -0,0 +1,1348 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+	"time"
+
+	snapV1 "github.com/kubernetes-csi/external-snapshotter/client/v6/apis/volumesnapshot/v1"
+	snapclient "github.com/kubernetes-csi/external-snapshotter/client/v6/clientset/versioned"
+	ginkgo "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	cnstypes "github.com/vmware/govmomi/cns/types"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	pbmtypes "github.com/vmware/govmomi/pbm/types"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fnodes "k8s.io/kubernetes/test/e2e/framework/node"
+	fpod "k8s.io/kubernetes/test/e2e/framework/pod"
+	fpv "k8s.io/kubernetes/test/e2e/framework/pv"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+/*
+	Test to verify sVmotion works fine for volumes in detached state
+
+	Steps
+	1. Create StorageClass.
+	2. Create PVC.
+	3. Expect PVC to pass and verified it is created correctly.
+	4. Relocate detached volume
+	5. Invoke CNS Query API and validate datastore URL value changed correctly
+*/
+
+var _ = ginkgo.Describe("[csi-block-vanilla] [csi-block-vanilla-parallelized] Relocate detached volume ", func() {
+	f := framework.NewDefaultFramework("svmotion-disk")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+	var (
+		client          clientset.Interface
+		namespace       string
+		scParameters    map[string]string
+		datastoreURL    string
+		sourceDatastore *object.Datastore
+		destDatastore   *object.Datastore
+		datacenter      *object.Datacenter
+		destDsURL       string
+		pvclaims        []*v1.PersistentVolumeClaim
+		fcdID           string
+		labelKey        string
+		labelValue      string
+	)
+	ginkgo.BeforeEach(func() {
+		bootstrap()
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+		scParameters = make(map[string]string)
+		datastoreURL = GetAndExpectStringEnvVar(envSharedDatastoreURL)
+		destDsURL = GetAndExpectStringEnvVar(destinationDatastoreURL)
+		nodeList, err := fnodes.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		labelKey = "app"
+		labelValue = "e2e-labels"
+	})
+
+	// Test for relocating volume being detached state
```

Comment:
detached state can also be volume attached to the pod first and then detached right?

Reply:
Not sure with what intention this testcase was written, this is not part of my changes.